mlir/include/mlir/Dialect/Rock/IR/MfmaInsnGroup.h

```diff
@@ -110,6 +110,14 @@ struct MfmaInsnGroupAttr {
   int64_t mRepeats;
   int64_t nRepeats;
   SmallVector<MFMAParams, 2> imms;
+  // Reduction constructor
+  MfmaInsnGroupAttr(const SmallString<16> &insn)
+      : insn{insn}, mRepeats{1}, nRepeats{1},
+        imms{{{0, 0, amdgpu::MFMAPermB::none}}} {}
+  // Broadcast constructor
+  MfmaInsnGroupAttr(const SmallString<16> &insn,
+                    const SmallVector<MFMAParams, 2> &imms)
+      : insn{insn}, mRepeats{1}, nRepeats{1}, imms{imms} {}
```

Comment:
The first is not possible, because 128 tiles are not in the table, but the user can still select them (and we would use the 64 entr, but mRepeats needs to be 128/32=4 instead of 64/32=2). I can get on with the second method you propose (I also quite like it)

Reply:
I still think the first approach is doable because at the initialization time we already know the wave size and the selected instruction. So it is equivalent to the computation you carried on below for each entry. But I agree that it may not worth the effort to move such trivial computation into the static initialization. Good choice on the second alternative~